### PR TITLE
[Merged by Bors] - CI for New System Tests

### PR DESCRIPTION
**Pull request by web3rover**
This PR is to run the new system tests in the CI. The new system tests are already merged to develop. It should run the tests during `bors try` and `bors merge` in our GCP kubernetes infra.

Tasks:

- [x]  It builds the go-spacemesh and system tests docker images and pushes to dockerhub
- [x]  It runs the tests in a k8s cluster hosted on GCP
- [x]  After running the tests it prints the tests results and cleans all pods
- [x]  Run system tests only when bors try or bors merge is invoked
- [x]  Fail the CI when tests fail (read the status of kubernetes pod)

### CODE REVIEW

**Comment by dshulyak** [comment]
@narayanprusty i merged pull request with tests. note that the module is `systest` (without s on the end).

File: .github/workflows/systest-ci.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      run: docker tag go-spacemesh-dev:${{ steps.vars.outputs.branch_name }} spacemeshos/go-spacemesh-dev:${{ steps.vars.outputs.sha_short }} && docker push spacemeshos/go-spacemesh-dev:${{ steps.vars.outputs.sha_short }}
+  
+    - name: Build tests docker image
+      run: cd systests && make docker
+
+    - name: Push tests docker images
+      run: cd systests && make push
+
+    - name: Run tests
+      run: cd systests && make run test_name=TestSmeshing
```
Comment (dshulyak) [inline .github/workflows/systest-ci.yml]: this is what i am using `make run test_name=. smesher_image=spacemeshos/go-spacemesh-dev:develop size=50 bootstrap=10m level=info clusters=2 node_selector=cloud.google.com/gke-nodepool=dmitry-ci-preemtible-1`

Comment (dshulyak) [reply .github/workflows/systest-ci.yml]: `clusters` controls how many clusters of `size` will be deployed concurrently. pool that is specified with `node-selector` should have enough resources to schedule pods on them.

File: systest/artifacts/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+helm repo add grafana https://grafana.github.io/helm-charts
+helm repo update
+
+helm install grafana grafana/grafana --values ./grafana.yaml
+helm install loki grafana/loki --values ./loki.yaml
+helm install promtail grafana/promtail --values ./promtail.yaml
```
Comment (dshulyak) [inline systest/artifacts/readme.md]: i was using chart `grafana/loki-stack ` 

```
helm upgrade --install loki grafana/loki-stack  --set grafana.enabled=true,prometheus.enabled=true,prometheus.alertmanager.persistentVolume.enabled=false,prometheus.server.persistentVolume.enabled=false,loki.persistence.enabled=true,loki.persistence.storageClassName=standard,loki.persistence.size=20Gi -f promtail.yaml
```

it automatically adds loki source to the grafana

what about cri-o pipeline, did you enable it or logs collector works without it?

**Review by dshulyak — COMMENTED** [review]
one general note is that artifacts should not be added to this repository, lets create a separate repo (for example `ci-infra`) with automation (using terraform as you mentioned)

File: systest/artifacts/readme.md (hunk 2 above)
Comment (web3rover) [reply systest/artifacts/readme.md]: The collector seems to work without the cri-o pipeline.

**Comment by web3rover** [comment]
@dshulyak I have removed artefacts from this repo. I have created a separate repo and added the artifacts there. I will create a terraform playbook which uses grafana/loki-stack in that repo.

**Comment by dshulyak** [comment]
@narayanprusty i realized that we should run new ci job only on `bors try` and `bors merge`. otherwise, it should be skipped as the rest of the long-running jobs.

**Comment by web3rover** [comment]
bors try

**Comment by narayanprusty** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -158,6 +158,73 @@ jobs:
...
+        LOGS="${LOGS//$'\r'/'%0D'}"
+        echo "::set-output name=content::$LOGS"
+      id: tests_logs
+    
+    - name: Delete pod
+      run: kubectl delete pod systest-${{ steps.vars.outputs.sha_short }}
+
+    - name: Tests Passed or Failed
+      run: |
+        LOGS='${{ steps.tests_logs.outputs.content }}'
```
Comment (dshulyak) [inline .github/workflows/ci.yml]: i think you should check pod state instead. if program exits with any non zero code pod state should be failed

Comment (dshulyak) [reply .github/workflows/ci.yml]: e.g. we need to fail when tests can't compile

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Review by dshulyak — APPROVED** [review]

**Review by moshababo — APPROVED** [review]

**Comment by web3rover** [comment]
bors merge

**Comment by web3rover** [comment]
bors merge

**Comment by web3rover** [comment]
bors try

**Pull request closed without merging**